pact/v3/ffi.py

```diff
@@ -16,26 +16,79 @@
 -   Basic Enum classes
 -   Simple wrappers around functions, including the casting of input and output
     values between the high level Python types and the low level C types.
+-   Simple wrappers around some of the low-level types. Specifically designed to
+    automatically handle the freeing of memory when the Python object is
+    destroyed.
 
 These low-level functions may then be combined into higher level classes and
-modules.
+modules. Ideally, all code outside of this module should be written in pure
+Python and not worry about allocating or freeing memory.
 
 During initial implementation, a lot of these functions will simply raise a
 `NotImplementedError`.
 
 For those unfamiliar with CFFI, please make sure to read the [CFFI
 documentation](https://cffi.readthedocs.io/en/latest/using.html).
+
+### Handles
+
+The Rust library exposes a number of handles to internal data structures. This
+is done to avoid exposing the internal implementation details of the library to
+users of the library, and avoid unnecessarily casting to and from possibly
+complicated structs.
+
+In the Rust library, the handles are thin wrappers around integers, and
```

Comment:
This was done to make it easier for Go and JS, but if it is a problem, we can change it to be a pointer type

Reply:
It's not at all an issue. It's just that a `struct { int }` has the same in-memory representation as a bare `int`. I added the thin wrappers for two reasons:

- Obfuscating the `int`. After all, it doesn't make sense to 'add' or 'subtract' handles (unlike integers). It also makes it explicit that the handles to a Pact or an Interaction are different.
- This makes it easy to implement the `__del__` method to ensure that memory is cleaned up when Python's garbage control executes.

Reply:
Great explanation @JP-Ellis 👍🏾 thank you!